### PR TITLE
add support for item resort API

### PR DESCRIPTION
**Pull request by wookiehangover**
#### What does it do?

Creates a method on item models to use the newly-landed resort api https://github.com/sprintly/sprint.ly/pull/1851

#### Where should the reviewer start?

`lib/items/item.js`

#### How should it be manually tested?

Checkout the reordering branch of sprintly/5columns and use `npm link` to test that the code in this branch correctly calls the API, moves items around. 

#### GIF?
6e9fabcf

### Should this trigger a version bump?
- [x] Yes
  * [ ] MAJOR
  * [x] MINOR
  * [ ] PATCH
- [ ] No

### CODE REVIEW

File: test/items/item-model-test.js [hunk 1]
```diff
@@ -104,6 +104,57 @@ describe('Item Model', function() {
...
+        [
+          200,
+          { 'Content-Type': 'application/json' },
+          '{ "sort": 20000 }'
+        ]
+      );
+
+      var req = this.item.resort({ before: 2 });
+      this.server.respond();
+      return req;
```
Comment (justinabrahms) [inline test/items/item-model-test.js]: Feels like there should be an assert here, unless I don't know something about how mocha handles returns (likely)?

Comment (wookiehangover) [inline test/items/item-model-test.js]: If you return a promise it will you it to pass / fail the test based on it's outcome

**Comment by wookiehangover** [comment]
Unless anyone objects, I'm going to land this and cut a new release to npm @justinabrahms @florapdx

**Comment by justinabrahms** [comment]
Fine with me.

**Pull request merged**